tree/treeplayer/test/basic.cxx

```diff
@@ -335,3 +338,36 @@ TEST(TTreeReaderBasic, InfLoop)
    TTreeReaderArray<float> x(tr, "0.2.0.energy");
    tr.Next();
 }
+
+// ROOT-10019
+TEST(TTreeReaderBasic, DisappearingBranch)
+{
+
+   auto createFile = [](const char *fileName, int ncols) {
+      // auto r = ROOT::RDataFrame(1).Define("col0",[](){return 0;}).Snapshot<int>("t","f1.root",{"col0"});
+      // r->Define("col1",[](){return 0;}).Snapshot<int,int>("t","f0.root",{"col0","col1"});
+      TFile f(fileName, "RECREATE");
```

Comment:
Could we use a TMemFile file here?

Reply:
Do you know how to add those to the chain and mimick the broken use case described in the JIRA item?

Reply:
humm looks like I was remembering wrong (i thought there was an interface to add either a TFile* or a TTree*).